apps/core/scenes/results/ResultsListItem.js

```diff
@@ -1,61 +1,195 @@
+        (localDeparture && DateFNS.format(localDeparture, timeSimpleFormat)) ??
+        '',
+      duration: duration ?? '',
+      id: (route && route.id) ?? '',
+    };
+  };
+
+  getSectorBorderIndex = () => {
+    const { route, routes } = this.props.data;
+    if (!route) {
+      return null;
+    }
+    return route.findIndex(
+      routeItem => routeItem?.departure?.cityCode === routes?.[1]?.[0],
+    );
+  };
+
+  // @TODO: make ConnectionCard more universal (not only for forth and back routes)
+  getSanitizedRoute = (): Array<TripSectorWithId> => {
+    const { route } = this.props.data;
+    if (route) {
+      return route.map((routeItem: RouteItem) =>
+        this.getSanitizedRouteItem(routeItem),
+      );
+    }
+    return [];
+  };
+
+  getSector = (
+    route: Array<TripSectorWithId>, // rename to sanitizeRoute
+    demandedSector: number,
+    sectorBorderIndex: ?number,
+  ): Array<TripSectorWithId> => {
+    if (sectorBorderIndex != null) {
+      if (demandedSector === 0) {
+        return route.slice(0, sectorBorderIndex);
+      }
+      return route.slice(sectorBorderIndex);
+    }
+    return route;
+  };
+
+  getDurationBetweenSectors = (route: ?Route, sectorBorderIndex: ?number) => {
+    if (route && sectorBorderIndex != null && sectorBorderIndex > 0) {
+      const stopBeforeSector = route[sectorBorderIndex - 1];
+      const stopAfterSector = route[sectorBorderIndex];
+      const utcDeparture = this.parseDate(stopAfterSector?.departure?.utcTime);
+      const utcArrival = this.parseDate(stopBeforeSector?.arrival?.utcTime);
+      return DateFNS.distanceInWords(utcDeparture, utcArrival);
+    }
+    return null;
+  };
+
+  render() {
+    const { data } = this.props;
+    const sanitizedRoute = this.getSanitizedRoute();
+    const sectorBorderIndex = this.getSectorBorderIndex();
+    const hasSectorBorder = sectorBorderIndex != null && sectorBorderIndex > -1;
+    if (data == null) {
+      return null;
+    }
+    const { price } = data;
+    const badges = [
+      {
+        id: '1',
+        type: 'warning',
+        children: 'Cheapest',
+      },
+      {
+        id: '2',
+        type: 'neutral',
+        children: 'Wi-fi',
+      },
+    ];
+    const priceObject = {
+      amount: parseFloat(price?.amount) ?? 0,
+      currency: price?.currency ?? 'CZK',
+    };
+
+    const duration =
+      this.getDurationBetweenSectors(data.route, sectorBorderIndex) ?? '';
+    const hasReturnFlight = sectorBorderIndex != null && sectorBorderIndex > 0;
+
+    // @TODO: use Card component
+    return (
+      <View style={styles.card}>
+        <ConnectionCard
+          wayForth={this.getSector(sanitizedRoute, 0, sectorBorderIndex)}
+          wayBack={
+            hasReturnFlight
+              ? this.getSector(sanitizedRoute, 1, sectorBorderIndex)
+              : undefined
+          }
+          badges={badges}
+          price={priceObject}
+          duration={hasSectorBorder ? duration : ''}
```

Comment:
Could you extract this logic before the return statement?

Reply:
OK